compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -252,6 +255,8 @@ struct SharedState<'tcx> {
     mentioned: MTLock<UnordSet<MonoItem<'tcx>>>,
     /// Which items are being used where, for better errors.
     usage_map: MTLock<UsageMap<'tcx>>,
+    /// Number of errors that have occurred during collection.
+    error_count: AtomicUsize,
```

Comment:
Is it correct to use `AtomicUsize` here or should we use an internal compiler version?